ml-agents/setup.py

```diff
@@ -71,13 +71,17 @@ def run(self):
         "cattrs>=1.0.0,<1.1.0",
         "attrs>=19.3.0",
         'pypiwin32==223;platform_system=="Windows"',
+        "importlib_metadata",  # TODO for python<3.8 only
     ],
     python_requires=">=3.6.1",
     entry_points={
         "console_scripts": [
             "mlagents-learn=mlagents.trainers.learn:main",
             "mlagents-run-experiment=mlagents.trainers.run_experiment:main",
-        ]
+        ],
+        "mlagents.stats_writer": [
```

Comment:
We would need to add a new key for each type of plugin interface we want, e.g. `mlagents.demonstration_provider: [...]`